### PR TITLE
Makes crc_cpp compatible with conan 2.x

### PR DESCRIPTION
**Pull request by Xav83**
Specify library name and version:  **crc_cpp/1.2.0**

Hello conan devs 🙂 

When trying to use the crc_cpp package with the latest version of conan (`2.0.9` at the moment), the following issue is raised:
```
ERROR: crc_cpp/1.2.0: Error in package_id() method, line 53
        self.info.header_only()
        AttributeError: 'ConanInfo' object has no attribute 'header_only'
```

This commit should correct this error, and is based on the elements mentionned in this [GitHub issue](https://github.com/conan-io/conan/issues/12689).

Thanks in advance for your help and your attention to this Pull Request
---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/crc_cpp/all/conanfile.py [hunk 1]
```diff
@@ -38,16 +40,16 @@ def _supported_compiler(self):
...
+            check_min_cppstd(self, "17")
         if not self._supported_compiler:
             raise ConanInvalidConfiguration("crc_cpp: Unsupported compiler: {}-{} "
                                             "Minimum C++17 constexpr features required.".format(self.settings.compiler, self.settings.compiler.version))
     def source(self):
-       tools.files.get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
 
     def package(self):
         self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
```
Comment (valgur) [inline recipes/crc_cpp/all/conanfile.py:51]: You will need to use full paths like dst=os.path.join(self.package_folder, "licenses") in Conan 2.x.

**Review by valgur — CHANGES_REQUESTED** [review]

**Review by uilianries — COMMENTED** [review]
@Xav83 Thank you for your contribution! Please, consider my PR https://github.com/Xav83/conan-center-index/pull/1 as an improvement over your changes!

File: recipes/crc_cpp/all/test_v1_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
```
Comment (uilianries) [inline recipes/crc_cpp/all/test_v1_package/conanfile.py:7]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"
```

Add missing cmake deps generator for Conan 1.x

**Comment by Xav83** [comment]
@uilianries , since #19912 has been merge, is the work in this Pull Request still relevant ?

**Comment by uilianries** [comment]
@Xav83 I don't think so 😞 I'll close this PR, but thank so much for contributing. Please, file an issue in case you find something in that merged PR.

**Pull request closed without merging**